src/Extension/Publishable/PublishableSiteTree.php

```diff
@@ -2,103 +2,186 @@
 
 namespace SilverStripe\StaticPublishQueue\Extension\Publishable;
 
+use SilverStripe\CMS\Model\RedirectorPage;
+use SilverStripe\CMS\Model\SiteTree;
 use SilverStripe\CMS\Model\VirtualPage;
 use SilverStripe\Control\Director;
 use SilverStripe\ORM\DataExtension;
+use SilverStripe\ORM\SS_List;
+use SilverStripe\SiteConfig\SiteConfig;
 use SilverStripe\StaticPublishQueue\Contract\StaticallyPublishable;
 use SilverStripe\StaticPublishQueue\Contract\StaticPublishingTrigger;
-use SilverStripe\CMS\Model\RedirectorPage;
+use SilverStripe\StaticPublishQueue\Extension\Engine\SiteTreePublishingEngine;
 
 /**
  * Bare-bones implementation of a publishable page.
  *
  * You can override this either by implementing one of the interfaces the class directly, or by applying
  * an extension via the config system ordering (inject your extension "before" the PublishableSiteTree).
  *
- * @TODO: re-implement optional publishing of all the ancestors up to the root? Currently it only republishes the parent
- *
  * @see SiteTreePublishingEngine
+ * @method SiteTree|SiteTreePublishingEngine|$this getOwner()
  */
 class PublishableSiteTree extends DataExtension implements StaticallyPublishable, StaticPublishingTrigger
 {
+    public const RELATION_INCLUDE_NONE = 0;
+    public const RELATION_INCLUDE_DIRECT = 1;
+    public const RELATION_INCLUDE_RECURSIVE = 2;
+
+    private static int $regenerate_children = self::RELATION_INCLUDE_NONE;
```

Comment:
Looks like this is only called a couple of times in the src code (as opposed to tests), and both times it's only checked for truthiness rather than as a boolean.  Would it make more sense to just makes this a bool instead?